tests/run_tests.py

```
@@ -2959,6 +2959,503 @@ def boom(path):
+    # E5-S1: a finding with no usable line, and a finding whose inline anchor is rejected (422),
+    # both fall back into the summary rather than being dropped.
+    pp = _import_pr_publish()
+    root = Path(tempfile.mkdtemp(prefix="ar-prpub-"))
+    try:
+        # no-line finding -> unanchored, never attempts an inline create
+        r1 = _write_run(root / "r1", reports={"correctness": [
+            {"id": "c1", "title": "NoLine", "severity": "medium", "file": "a.py", "line": None}]})
+        gh1 = _FakeGitHub(pp.GitHubError)
+        p1 = pp.publish(r1, {"repo": "o/r", "pr": 2, "commit_sha": "s", "fail_on": "blocked"}, gh1)
+        assert p1["created"] == 0 and p1["unanchored"] == 1 and len(gh1.review) == 0, p1
+        summary = list(gh1.issue.values())[0]["body"]
```

Ruff (0.16.1)

[warning] Prefer `next(iter(gh1.issue.values()))` over single element slice

Replace with `next(iter(gh1.issue.values()))`

(RUF015)